### PR TITLE
feat(sync) add UUID logic.

### PR DESCRIPTION
**Pull request by Enuviel**

### CODE REVIEW

**Comment by benmarten** [comment]
test this please

File: AndroidSDK/src/com/leanplum/internal/Request.java [hunk 1]
```diff
@@ -383,11 +384,13 @@ private void sendNow() {
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        final List<Map<String, Object>> requestsToSend = getUnsentRequests();
+        List<Map<String, Object>> requestsToSend = getUnsentRequests();
         if (requestsToSend.isEmpty()) {
           return null;
         }
 
+        requestsToSend = updateUnsentRequests(requestsToSend);
```
Comment (milos1290) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: why do we need to update unsent requests ? couldn't we generate uuid when inserting ? This just causes double reading/parsing and saving again.

Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: I agree, can the uuid be hold in shared preferences, until next send, and included during inserting of events....
only on sending the events, we will update the uuid?

Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/internal/Request.java]: we should be sure that all requests were saved with generated UUID before sending them to the server.
We will have one UUID for the batch of actions, that means we cannot generate UUID when inserting. We can avoid double parsing with moving UUID logic to getUnsentRequests.

**Pull request merged**